### PR TITLE
Date fields

### PR DESCRIPTION
**Pull request by mgax**
closes https://github.com/hoover/snoop2/issues/45

### CODE REVIEW

File: snoop/data/analyzers/tika.py [hunk 1]
```diff
@@ -71,3 +72,22 @@ def rmeta(blob):
...
+
+
+def get_date_created(rmeta):
+    FIELDS_CREATED = ['Creation-Date', 'dcterms:created', 'meta:created',
+                      'created']
+
+    for field in FIELDS_CREATED:
+        value = rmeta[0].get(field)
+        if value:
+            return parser.parse(value).isoformat()
```
Comment (gabriel-v) [inline snoop/data/analyzers/tika.py]: `utils.zulu` format?

File: snoop/data/analyzers/tika.py [hunk 2]
```diff
@@ -71,3 +72,22 @@ def rmeta(blob):
...
+
+def get_date_modified(rmeta):
+    FIELDS_MODIFIED = ['Last-Modified', 'Last-Saved-Date', 'dcterms:modified',
+                       'meta:modified', 'created']
+
+    for field in FIELDS_MODIFIED:
+        value = rmeta[0].get(field)
+        if value:
+            return parser.parse(value).isoformat()
```
Comment (gabriel-v) [inline snoop/data/analyzers/tika.py]: `utils.zulu` format?

File: snoop/data/digests.py [hunk 3]
```diff
@@ -134,12 +136,18 @@ def iter_parts(email_data):
     for header in ['To', 'Cc', 'Bcc', 'Resent-To', 'Recent-Cc']:
         headers_to.update(headers.get(header, []))
 
+    message_date = None
+    message_raw_date = headers.get('Date', [None])[0]
+    if message_raw_date:
+        message_date = email.parse_date(message_raw_date).isoformat()
```
Comment (gabriel-v) [inline snoop/data/digests.py]: `utils.zulu` format?

**Comment by mgax** [comment]
> Why not use the utils.zulu format

Yeah good catch.

**Comment by gabriel-v** [comment]
You missed a few `+00:00` in your tests I think.

But we have another problem: that `zulu` function does assert `date.endswith('+00:00')`. What if we get a date with a specific timezone from tika or the email file?

Can we fix & test the `zulu` function so if the date given to it has a timezone, it gets changed to UTC?

**Comment by mgax** [comment]
Hmm, `zulu` only works for timestaps in the UTC timezone. And we don't have to use it, any iso-formatted date should work fine with elasticsearch. Reverting the last patch.

**Comment by gabriel-v** [comment]
Then why are we using `zulu` anywhere?

**Comment by mgax** [comment]
Can't remember. Was it because ES didn't always auto-detect them as dates?

**Comment by mgax** [comment]
There: https://github.com/hoover/snoop2/blob/52665c93208a96c3ed968b2582a4821a5b3fbbbd/snoop/data/utils.py#L33

**Comment by gabriel-v** [comment]
Looking at the docs, yeah, timezones should be supported.

Do run some checks about it, and look at https://www.elastic.co/guide/en/elasticsearch/reference/2.4/date.html which says they turn all dates into UTC anyway. So we might as well be safe and convert them to the UTC Z format as they want it.

**Comment by gabriel-v** [comment]
travis: `NameError("name 'zulu' is not defined",)` in the integration test

**Comment by mgax** [comment]
I made some tests, ES indexes them properly, and filtering by date/time range works.

**Pull request merged**